hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java

```diff
@@ -404,6 +407,11 @@ public void run() {
           "This cycle terminating immediately because 'shouldRun' has been deactivated");
       return;
     }
+    if (reconcileRunning.get()) {
```

Comment:
Using the `reconcileRunning` variable in this way cannot prevent two threads from executing `reconcile()` at the same time. When this thread is running after line410 and before line416, another thread may set `reconcileRunning` to true.

Reply:
@zhangshuyan0 Hi, sir. Here should use synchronized to protect. I forgot to push. Will push it after vocation.  Thanks for your review.